javascript/javascript1/week2/nameGender.js

```diff
@@ -0,0 +1,55 @@
+let fullName;
+let firstName;
+let surName;
+let useFormalName;
+let isMale;
+
+function getFullname(firstName, surName, useFormalName, isMale) {
+    if ((firstName == null) || (firstName == '') || (firstName.indexOf(' ') >= 0) ){
+        return fullName = `Name value is not entered properly`;
+    }
+
+    else if ((!useFormalName) && (!surName)){
+        return fullName = `${firstName}`;
+    }
+    else if ((!useFormalName) && (!isMale)){
+        return fullName = `${firstName} ${surName}`;
+    }
+    else if ((useFormalName) && (!surName) && (isMale)){
+            if (isMale) {
+            return fullName = `Lord ${firstName}`;
+            }
+            else
+            return fullName = `Highness ${firstName}`;
+    }
+    else if (isMale) {
+            return fullName = `Lord ${firstName} ${surName}`;
+        }
+        else
+        return fullName = `Highness ${firstName} ${surName}`;
+    };
```

Comment:
This would be simpler without the chain of if/else-statements, I think:
```js
if (!firstName || firstName.indexOf(' ') >= 0) {
	return fullName = `Name value is not entered properly`;
}

let formalName = ""
if (useFormalName) {
	if (isMale) {
		formalName = "Lord "
	} else {
		formalName = "Highness "
	}
}

if (surName) {
	return fullName = `${formalName}${firstName} ${surName}`
} else {
	return fullName = `${formalName}${firstName}`
}
```